### PR TITLE
add parcel repository manifest creator

### PR DESCRIPTION
**Pull request by dereklwood**
Port of Cloudera's [parcel repository manifest creator](https://github.com/cloudera/cm_ext/tree/master/make_manifest), to get around the Python 2.7 requirement.
   - [x] all requires are part of stdlib or rubygems... no external gems needed
   - [x] output only differs by hash key ordering

```
$ ./make_manifest.rb ~/git/cm_ext/make_manifest/test_artifacts/
Scanning directory: /Users/derek/git/cm_ext/make_manifest/test_artifacts/
Found parcel CDH-5.0.0-0.cdh5b2.p0.30-precise.parcel

$ head -20 ~/git/cm_ext/make_manifest/test_artifacts/manifest.json 
{
    "lastUpdated": 1452908566,
    "parcels": [
        {
            "parcelName": "CDH-5.0.0-0.cdh5b2.p0.30-precise.parcel",
            "hash": "fa95c0f7e930f7363118f187f22685d0299d036a",
            "replaces": "IMPALA, SOLR, SPARK",
            "components": [
                {
                    "name": "bigtop-tomcat",
                    "version": "6.0.37-cdh5.0.0-beta-2",
                    "pkg_version": "0.7.0+cdh5.0.0+0"
                },
                {
                    "name": "crunch",
                    "version": "0.9.0-cdh5.0.0-beta-2",
                    "pkg_version": "0.9.0+cdh5.0.0+19"
                },
                {
                    "name": "flume-ng",

$ tail ~/git/cm_ext/make_manifest/test_artifacts/manifest.json 
                {
                    "name": "zookeeper",
                    "version": "3.4.5-cdh5.0.0-beta-2",
                    "pkg_version": "3.4.5+cdh5.0.0+27"
                }
            ],
            "releaseNotes": "Here are some release notes.\n\nWith new lines and shit.\n"
        }
    ]
```

### CODE REVIEW

File: cdap-distributions/bin/.rubocop_todo.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Offense count: 1
+Style/AsciiComments:
+  Exclude:
+    - 'make_manifest.rb'
+
+# Offense count: 1
+# Cop supports --auto-correct.
+Style/LeadingCommentSpace:
+  Exclude:
+    - 'make_manifest.rb'
```
Comment (wolf31o2) [inline cdap-distributions/bin/.rubocop_todo.yml]: Should we move these files to the top of the repo and maintain them as repo global?

Comment (dereklwood) [inline cdap-distributions/bin/.rubocop_todo.yml]: On one hand I don't expect we'll be adding much Ruby to this repo (and this script likely won't change).  On the other hand it's probably the right thing to do.  @sreevatsanraman any thoughts?

**Comment by dereklwood** [comment]
I need to add headers for RAT check... standby for that

**Comment by anwar6953** [comment]
We don't need this in release/3.3?

**Comment by wolf31o2** [comment]
It will need to be backported to multiple branches.

File: cdap-distributions/bin/.rubocop_todo.yml (hunk 1 above)
Comment (sreevatsanraman) [inline cdap-distributions/bin/.rubocop_todo.yml]: Makes sense to be on top level with other checkstyle file. This way it will be more organized.

Comment (dereklwood) [inline cdap-distributions/bin/.rubocop_todo.yml]: moved to top-level

**Comment by dereklwood** [comment]
rubocop files moved, and RAT exclusions added

**Comment by wolf31o2** [comment]
:+1: LGTM

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# on 2016-01-15 17:31:44 -0800 using RuboCop version 0.35.1.
+# The point is for the user to remove these configuration records
+# one by one as the offenses are removed from the code base.
+# Note that changes in the inspected code, or installation of new
+# versions of RuboCop, may require this file to be generated again.
+
+# Offense count: 1
+Lint/HandleExceptions:
+  Exclude:
+    - 'make_manifest.rb'
```
Comment (wolf31o2) [inline .rubocop_todo.yml]: These paths need to be fixed.

**Comment by dereklwood** [comment]
https://builds.cask.co/browse/CDAP-DUT3439-3/

merging and will backport

**Pull request merged**